apollo-router/src/query_planner/execution.rs

```diff
@@ -460,7 +519,7 @@ impl DeferredNode {
                     // or because it is lagging, but here we only send one message so it
                     // will not happen
                     if let Some(Ok((deferred_value, err))) = v {
-                        value.deep_merge(deferred_value);
+                        PlanNode::type_aware_deep_merge(&sc, &mut value, deferred_value);
```

Comment:
You missed replacing a `deep_merge()` with a type-aware merge after this usage on line 502/in `DeferredNode::execute()`.